packages/@lwc/template-compiler/src/codegen/helpers.ts

```diff
@@ -163,6 +163,31 @@ export function generateTemplateMetadata(codeGen: CodeGen): t.Statement[] {
     return metadataExpressions;
 }
 
+/**
+ * @todo: probably a better location is module.ts.
+ * @param codeGen
+ */
+export function generateApisInitialization(codeGen: CodeGen): t.VariableDeclaration[] {
+    if (Object.keys(codeGen.usedApis).length === 0) {
+        return [];
+    }
+
+    codeGen.usedLwcApis.add(RENDER_API);
+
+    return [
+        t.variableDeclaration('const', [
+            t.variableDeclarator(
+                t.objectPattern(
+                    Object.keys(codeGen.usedApis).map((name) =>
+                        t.assignmentProperty(t.identifier(name), codeGen.usedApis[name])
+                    )
+                ),
+                t.identifier(RENDER_API)
+            ),
+        ]),
+    ];
```

Comment:
From a tree-shaking perspective, it is probably better to export each render API independently. This would also remove the need to use an object spread at the top of each generated template. That said I am wondering if the tree-shaking benefits are outweighed by the increased output size for AMD.

Reply:
It also adds some unnecessary boilerplate in compat mode:

```js
lwc.renderApi._ES5ProxyType ? lwc.renderApi.get("d") : lwc.renderApi.d
```

> That said I am wondering if the tree-shaking benefits are outweighed by the increased output size for AMD.

I don't understand this. Could you explain?

Reply:
> > That said I am wondering if the tree-shaking benefits are outweighed by the increased output size for AMD.
> 
> I don't understand this. Could you explain?

Good catch @nolanlawson. I was mistaken, there is no increase in AMD module size. I was under the impression that using named imports would negatively impact the AMD module size by adding new entries in the dependency list (the first argument of the `define` function). But it's wrong, the list contains the dependency module names and not the named imports for each dependency.

```js
import { createVElement, createVText, createVComponent } from 'lwc';

createVElement();
createVText();
createVComponent();
```

```js
define(['lwc'], (function (lwc) { 'use strict';
	lwc.createVElement();
	lwc.createVText();
	lwc.createVComponent();
}));
```